salt/modules/smtp.py

```diff
@@ -106,22 +107,29 @@ def send_msg(
         sender = creds.get("smtp.sender")
         username = creds.get("smtp.username")
         password = creds.get("smtp.password")
+        port = creds.get("smtp.port", 0)
```

Comment:
If it's *only* about SMTP relay, it should be obviously `25`, did you test it with `467`, `587` etc? You might want to add missing tests as well

Reply:
I'm yet to test with 465 / 587. Here, when the default of 0 is used, the connection is made over the defaults (465 for SSL and 25 for non-ssl)

Reply:
About the tests I'm not well versed with, but could you let me know if unit tests only will do or do we need integration test as well ?

Reply:
I would add the tests into `tests/pytests/unit/states/test_smtp.py` and `tests/unit/modules/test_smtp.py`

Reply:
you might noticed the failing CI as well

Reply:
does the failing CI has something to do with tests ? Or is it something else ?

Reply:
> does the failing CI has something to do with tests ? Or is it something else ?

yep, they are related to this PR:
https://jenkins.saltproject.io/job/pr-debian-11-amd64-py3-pytest/job/PR-61504/1/testReport/junit/tests.unit.modules.test_smtp/SmtpTestCase/test_send_msg/

Reply:
I see, that is possibly because I've not added the newly added port property to the mock of config options, under tests/unit/modules/test_smtp.py - Adding that should solve this.

Reply:
**question** should that not fall back to `port`? i.e. line 88 defines port as 0 and here on 111 we're redefining that to 0 even if port was passed in as 45 or something.